libbeat/publisher/queue/memqueue/broker.go

```diff
@@ -108,15 +110,23 @@ func NewQueue(
 	logger logger,
 	settings Settings,
 ) queue.Queue {
-	// define internal channel size for producer/client requests
-	// to the broker
-	chanSize := 20
-
 	var (
 		sz           = settings.Events
 		minEvents    = settings.FlushMinEvents
 		flushTimeout = settings.FlushTimeout
 	)
+	// define internal channel size for producer/client requests
+	// to the broker. This is 20 by default, but the value can be
+	// increased to up to 10% of max numbers of events in queue.
+	const minIntQueueSize = 20
```

Comment:
This looks reasonable but could we make the `chanSize` calculation a helper function so we can add a simple unit test?